cdap-common/src/main/java/co/cask/cdap/common/utils/TimeMathParser.java

```diff
@@ -80,19 +81,27 @@ public static int resolutionInSeconds(String resolutionStr) {
   }
 
   public static long parseTime(String timeStr) {
-    return parseTime(nowInSeconds(), timeStr);
+    return parseTime(timeStr, TimeUnit.SECONDS);
+  }
+
+  public static long parseTime(String timeStr, TimeUnit timeUnit) {
+    return parseTime(nowInSeconds(), timeStr, timeUnit);
   }
 
   public static long parseTime(long now, String timeStr) {
+    return parseTime(now, timeStr, TimeUnit.SECONDS);
+  }
+
+  public static long parseTime(long now, String timeStr, TimeUnit timeUnit) {
     Preconditions.checkNotNull(timeStr);
 
     if (NOW.equals(timeStr.toUpperCase())) {
       return now;
     }
-    // if its a timestamp in seconds
+    // if its a numeric timestamp
     Matcher matcher = TIMESTAMP_PATTERN.matcher(timeStr);
     if (matcher.matches()) {
-      return Integer.parseInt(timeStr);
+      return Long.parseLong(timeStr);
```

Comment:
Why it ignores the time unit? Does it mean the string representation of the numeric value should be in the time unit as being passed in? Can you describe it in the javadoc of the method?